[Bug]: The terraform-provider-ncloud_v4.0.4.exe plugin crashed

### Terraform Core Version

1.14.0

### NCLOUD Provider Version

4.0.4

### Affected Resource(s)

[data source]
* ncloud_mongodb_products
* ncloud_mysql_products
* ncloud_mssql_products
* ncloud_postgresql_products
* ncloud_redis_products

### Expected Behavior

All database product data sources should successfully return G3 database product information when using G3 image product codes, similar to how they work with G2 product codes.

### Actual Behavior

**All G3 database product data sources crash** with identical nil pointer dereference panics. This appears to be a **systematic issue with G3 product handling** across the entire database product family, not just MongoDB.

- G2 products work correctly 
- G3 products consistently crash with the same stack trace pattern

### Relevant Error/Panic Output Snippet

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x101d057]

goroutine 12 [running]:
github.com/terraform-providers/terraform-provider-ncloud/internal/service/mongodb.(*mongodbProductModel).refreshFromOutput(0x0?, {0x0?, 0x2195340?}, 0x0?)
        github.com/terraform-providers/terraform-provider-ncloud/internal/service/mongodb/mongodb_products_data_source.go:254 +0x1f7
github.com/terraform-providers/terraform-provider-ncloud/internal/service/mongodb.flattenMongoDbProductLists({0x178fb68, 0xc0005408d0}, {0xc000438200, 0x31, 0x0?})
        github.com/terraform-providers/terraform-provider-ncloud/internal/service/mongodb/mongodb_products_data_source.go:185 +0xc5
```

### Terraform Configuration Files

```hcl
# Mongo DB
## This configuration works fine (G2 products)
data "ncloud_mongodb_products" "all_products_g2" {
    image_product_code = "SW.VMGDB.OS.LNX64.ROCKY.0810.MNGDB.CE.B050"
}

output "g2_products_count" {
    description = "G2 Community 전체 products 개수"
    value = length(data.ncloud_mongodb_products.all_products_g2.product_list)
}

## This configuration causes provider crash (G3 products)
data "ncloud_mongodb_products" "all_products_g3" {
    image_product_code = "SW.VMGDB.OS.LNX64.ROCKY.08.MNGDB.EE.B050.G003"
}

output "g3_products_count" {
    description = "G3 Community 전체 products 개수"
    value = length(data.ncloud_mongodb_products.all_products_g3.product_list)
}

# MySQL
data "ncloud_mysql_products" "this" {
    image_product_code = "SW.VMYSL.OS.LNX64.ROCKY.08.MYSQL.B050.G003"
}

output "ncloud_mysql_products_result" {
    value = data.ncloud_mysql_products.this
}

# MSSQL
data "ncloud_mssql_products" "this" {
    image_product_code = "SW.VMSSL.OS.WND64.WINNT.SVR2019.MSSQL.SE.B100.G003"
}

output "ncloud_mssql_products_result" {
    value = data.ncloud_mssql_products.this
}

# PostgreSQL
data "ncloud_postgresql_products" "this" {
    image_product_code = "SW.VPGSL.OS.LNX64.ROCKY.08.PGSQL.B050.G003"
}

output "ncloud_postgresql_products" {
    value = data.ncloud_postgresql_products.this
}

# Redis
data "ncloud_redis_products" "this" {
    redis_image_product_code = "SW.VRDS.OS.LNX64.ROCKY.08.REDIS.B050.G003"
}

output "ncloud_redis_products" {
    value = data.ncloud_redis_products.this
}
```

### Steps to Reproduce

1. Create a Terraform configuration with any G3 database product data source
2. Use any G3 image product code (pattern: `*.G003`)
3. Run `terraform plan`
4. Provider crashes with identical nil pointer dereference across all database types

**Affected G3 Product Codes:**
- MongoDB: `SW.VMGDB.OS.LNX64.ROCKY.08.MNGDB.EE.B050.G003`
- MySQL: `SW.VMYSL.OS.LNX64.ROCKY.08.MYSQL.B050.G003`  
- MSSQL: `SW.VMSSL.OS.WND64.WINNT.SVR2019.MSSQL.SE.B100.G003`
- PostgreSQL: `SW.VPGSL.OS.LNX64.ROCKY.08.PGSQL.B050.G003`
- Redis: `SW.VRDS.OS.LNX64.ROCKY.08.REDIS.B050.G003`

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

- **This is a systematic G3 issue, not MongoDB-specific**
- ALL G3 database product data sources fail with identical nil pointer crashes
- G2 products work perfectly across all database types
- The issue appears to be in the shared G3 product response processing logic
- **Critical Impact**: G3 infrastructure cannot be provisioned via Terraform
- Pattern suggests G3 API responses have different structure or contain unexpected nil values that aren't handled properly


### References

- Original stack trace shows crash in MongoDB `mongodb_products_data_source.go:254`
- Similar crashes likely occur in corresponding files for other database types
- This suggests a shared code pattern or G3-specific API response handling issue across all database services

### Would you like to implement a fix?

None

--- a/internal/service/hadoop/hadoop_image_products_data_source.go
+++ b/internal/service/hadoop/hadoop_image_products_data_source.go
@@ -185,8 +185,12 @@ func (h *hadoopImageProduct) refreshFromOutput(output *vhadoop.Product) {
 	h.ProductCode = types.StringPointerValue(output.ProductCode)
 	h.GenerationCode = types.StringPointerValue(output.GenerationCode)
 	h.ProductName = types.StringPointerValue(output.ProductName)
-	h.ProductType = types.StringPointerValue(output.ProductType.Code)
-	h.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	if output.ProductType != nil {
+		h.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
+	if output.PlatformType != nil {
+		h.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	}
 	h.OsInformation = types.StringPointerValue(output.OsInformation)
 	h.EngineVersionCode = types.StringPointerValue(output.EngineVersionCode)
 }

--- a/internal/service/hadoop/hadoop_products_data_source.go
+++ b/internal/service/hadoop/hadoop_products_data_source.go
@@ -239,11 +239,19 @@ func (_ hadoopProductModel) attrTypes() map[string]attr.Type {
 func (h *hadoopProductModel) refreshFromOutput(output *vhadoop.Product) {
 	h.ProductName = types.StringPointerValue(output.ProductName)
 	h.ProductCode = types.StringPointerValue(output.ProductCode)
-	h.ProductType = types.StringPointerValue(output.ProductType.Code)
+	if output.ProductType != nil {
+		h.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
 	h.ProductDescription = types.StringPointerValue(output.ProductDescription)
-	h.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
-	h.InfraResourceDetailType = types.StringPointerValue(output.InfraResourceDetailType.Code)
+	if output.InfraResourceType != nil {
+		h.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	}
+	if output.InfraResourceDetailType != nil {
+		h.InfraResourceDetailType = types.StringPointerValue(output.InfraResourceDetailType.Code)
+	}
 	h.CpuCount = common.Int64ValueFromInt32(output.CpuCount)
 	h.MemorySize = types.Int64PointerValue(output.MemorySize)
-	h.DiskType = types.StringPointerValue(output.DiskType.Code)
+	if output.DiskType != nil {
+		h.DiskType = types.StringPointerValue(output.DiskType.Code)
+	}
 }

--- a/internal/service/mongodb/mongodb_image_products_data_source.go
+++ b/internal/service/mongodb/mongodb_image_products_data_source.go
@@ -187,8 +187,12 @@ func (m *mongodbImageProduct) refreshFromOutput(output *vmongodb.Product) {
 	m.ProductCode = types.StringPointerValue(output.ProductCode)
 	m.GenerationCode = types.StringPointerValue(output.GenerationCode)
 	m.ProductName = types.StringPointerValue(output.ProductName)
-	m.ProductType = types.StringPointerValue(output.ProductType.Code)
-	m.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	if output.ProductType != nil {
+		m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
+	if output.PlatformType != nil {
+		m.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	}
 	m.OsInformation = types.StringPointerValue(output.OsInformation)
 	m.EngineVersionCode = types.StringPointerValue(output.EngineVersionCode)
 }

--- a/internal/service/mongodb/mongodb_products_data_source.go
+++ b/internal/service/mongodb/mongodb_products_data_source.go
@@ -245,11 +245,19 @@ func (m mongodbProductModel) attrTypes() map[string]attr.Type {
 func (m *mongodbProductModel) refreshFromOutput(ctx context.Context, output *vmongodb.Product) {
 	m.ProductCode = types.StringPointerValue(output.ProductCode)
 	m.ProductName = types.StringPointerValue(output.ProductName)
-	m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	if output.ProductType != nil {
+		m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
 	m.ProductDescription = types.StringPointerValue(output.ProductDescription)
-	m.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
-	m.InfraResourceDetailType = types.StringPointerValue(output.InfraResourceDetailType.Code)
+	if output.InfraResourceType != nil {
+		m.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	}
+	if output.InfraResourceDetailType != nil {
+		m.InfraResourceDetailType = types.StringPointerValue(output.InfraResourceDetailType.Code)
+	}
 	m.CpuCount = common.Int64ValueFromInt32(output.CpuCount)
 	m.MemorySize = types.Int64PointerValue(output.MemorySize)
-	m.DiskType = types.StringPointerValue(output.DiskType.Code)
+	if output.DiskType != nil {
+		m.DiskType = types.StringPointerValue(output.DiskType.Code)
+	}
 }

--- a/internal/service/mssql/mssql_image_products_data_source.go
+++ b/internal/service/mssql/mssql_image_products_data_source.go
@@ -207,7 +207,11 @@ func (m *mssqlImageProduct) refreshFromOutput(output *vmssql.Product) {
 	m.ProductCode = types.StringPointerValue(output.ProductCode)
 	m.GenerationCode = types.StringPointerValue(output.GenerationCode)
 	m.ProductName = types.StringPointerValue(output.ProductName)
-	m.ProductType = types.StringPointerValue(output.ProductType.Code)
-	m.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	if output.ProductType != nil {
+		m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
+	if output.PlatformType != nil {
+		m.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	}
 	m.OsInformation = types.StringPointerValue(output.OsInformation)
 }

--- a/internal/service/mssql/mssql_products_data_source.go
+++ b/internal/service/mssql/mssql_products_data_source.go
@@ -223,10 +223,16 @@ func (m mssqlProductModel) attrTypes() map[string]attr.Type {
 func (m *mssqlProductModel) refreshFromOutput(output *vmssql.Product) {
 	m.ProductCode = types.StringPointerValue(output.ProductCode)
 	m.ProductName = types.StringPointerValue(output.ProductName)
-	m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	if output.ProductType != nil {
+		m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
 	m.ProductDescription = types.StringPointerValue(output.ProductDescription)
-	m.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	if output.InfraResourceType != nil {
+		m.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	}
 	m.CpuCount = common.Int64ValueFromInt32(output.CpuCount)
 	m.MemorySize = types.Int64PointerValue(output.MemorySize)
-	m.DiskType = types.StringPointerValue(output.DiskType.Code)
+	if output.DiskType != nil {
+		m.DiskType = types.StringPointerValue(output.DiskType.Code)
+	}
 }

--- a/internal/service/mysql/mysql_image_products_data_source.go
+++ b/internal/service/mysql/mysql_image_products_data_source.go
@@ -185,8 +185,12 @@ func (m *mysqlImageProduct) refreshFromOutput(output *vmysql.CloudDbProduct) {
 	m.ProductCode = types.StringPointerValue(output.ProductCode)
 	m.GenerationCode = types.StringPointerValue(output.GenerationCode)
 	m.ProductName = types.StringPointerValue(output.ProductName)
-	m.ProductType = types.StringPointerValue(output.ProductType.Code)
-	m.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	if output.ProductType != nil {
+		m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
+	if output.PlatformType != nil {
+		m.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	}
 	m.OsInformation = types.StringPointerValue(output.OsInformation)
 	m.EngineVersionCode = types.StringPointerValue(output.EngineVersionCode)
 }

--- a/internal/service/mysql/mysql_products_data_source.go
+++ b/internal/service/mysql/mysql_products_data_source.go
@@ -223,10 +223,16 @@ func (m mysqlProductModel) attrTypes() map[string]attr.Type {
 func (m *mysqlProductModel) refreshFromOutput(output *vmysql.CloudDbProduct) {
 	m.ProductCode = types.StringPointerValue(output.ProductCode)
 	m.ProductName = types.StringPointerValue(output.ProductName)
-	m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	if output.ProductType != nil {
+		m.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
 	m.ProductDescription = types.StringPointerValue(output.ProductDescription)
-	m.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	if output.InfraResourceType != nil {
+		m.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	}
 	m.CpuCount = common.Int64ValueFromInt32(output.CpuCount)
 	m.MemorySize = types.Int64PointerValue(output.MemorySize)
-	m.DiskType = types.StringPointerValue(output.DiskType.Code)
+	if output.DiskType != nil {
+		m.DiskType = types.StringPointerValue(output.DiskType.Code)
+	}
 }

--- a/internal/service/postgresql/postgresql_image_products_data_source.go
+++ b/internal/service/postgresql/postgresql_image_products_data_source.go
@@ -194,8 +194,12 @@ func (d postgresqlImageProduct) attrTypes() map[string]attr.Type {
 func (d *postgresqlImageProduct) refreshFromOutput(output *vpostgresql.Product) {
 	d.ProductCode = types.StringPointerValue(output.ProductCode)
 	d.ProductName = types.StringPointerValue(output.ProductName)
-	d.ProductType = types.StringPointerValue(output.ProductType.Code)
-	d.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	if output.ProductType != nil {
+		d.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
+	if output.PlatformType != nil {
+		d.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	}
 	d.OsInformation = types.StringPointerValue(output.OsInformation)
 	d.GenerationCode = types.StringPointerValue(output.GenerationCode)
 	d.EngineVersionCode = types.StringPointerValue(output.EngineVersionCode)

--- a/internal/service/postgresql/postgresql_products_data_source.go
+++ b/internal/service/postgresql/postgresql_products_data_source.go
@@ -234,10 +234,16 @@ func (d postgresqlProductModel) attrTypes() map[string]attr.Type {
 func (d *postgresqlProductModel) refreshFromOutput(output *vpostgresql.Product) {
 	d.ProductCode = types.StringPointerValue(output.ProductCode)
 	d.ProductName = types.StringPointerValue(output.ProductName)
-	d.ProductType = types.StringPointerValue(output.ProductType.Code)
+	if output.ProductType != nil {
+		d.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
 	d.ProductDescription = types.StringPointerValue(output.ProductDescription)
-	d.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	if output.InfraResourceType != nil {
+		d.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	}
 	d.CpuCount = common.Int64ValueFromInt32(output.CpuCount)
 	d.MemorySize = types.Int64PointerValue(output.MemorySize)
-	d.DiskType = types.StringPointerValue(output.DiskType.Code)
+	if output.DiskType != nil {
+		d.DiskType = types.StringPointerValue(output.DiskType.Code)
+	}
 }

--- a/internal/service/redis/redis_image_products_data_source.go
+++ b/internal/service/redis/redis_image_products_data_source.go
@@ -185,8 +185,12 @@ func (r *redisImageProduct) refreshFromOutput(output *vredis.Product) {
 	r.ProductCode = types.StringPointerValue(output.ProductCode)
 	r.GenerationCode = types.StringPointerValue(output.GenerationCode)
 	r.ProductName = types.StringPointerValue(output.ProductName)
-	r.ProductType = types.StringPointerValue(output.ProductType.Code)
-	r.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	if output.ProductType != nil {
+		r.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
+	if output.PlatformType != nil {
+		r.PlatformType = types.StringPointerValue(output.PlatformType.Code)
+	}
 	r.OsInformation = types.StringPointerValue(output.OsInformation)
 	r.EngineVersionCode = types.StringPointerValue(output.EngineVersionCode)
 }

--- a/internal/service/redis/redis_products_data_source.go
+++ b/internal/service/redis/redis_products_data_source.go
@@ -225,10 +225,16 @@ func (r redisProductModel) attrTypes() map[string]attr.Type {
 func (r *redisProductModel) refreshFromOutput(output *vredis.Product) {
 	r.ProductCode = types.StringPointerValue(output.ProductCode)
 	r.ProductName = types.StringPointerValue(output.ProductName)
-	r.ProductType = types.StringPointerValue(output.ProductType.Code)
+	if output.ProductType != nil {
+		r.ProductType = types.StringPointerValue(output.ProductType.Code)
+	}
 	r.ProductDescription = types.StringPointerValue(output.ProductDescription)
-	r.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	if output.InfraResourceType != nil {
+		r.InfraResourceType = types.StringPointerValue(output.InfraResourceType.Code)
+	}
 	r.CpuCount = common.Int64ValueFromInt32(output.CpuCount)
 	r.MemorySize = types.Int64PointerValue(output.MemorySize)
-	r.DiskType = types.StringPointerValue(output.DiskType.Code)
+	if output.DiskType != nil {
+		r.DiskType = types.StringPointerValue(output.DiskType.Code)
+	}
 }

